src/nupic/algorithms/Anomaly.hpp

```diff
@@ -112,10 +113,16 @@ namespace nupic
                        const std::vector<UInt>& predicted,
                        Real64 inputValue=0, UInt timestamp=0);
 
+//helper
+        void setLikelihood(AnomalyLikelihood newInstance) {
+          this->likelihood_ = newInstance;
+        }
+
       private:
         AnomalyMode mode_;
         Real32 binaryThreshold_;
         std::unique_ptr<nupic::util::MovingAverage> movingAverage_;
+        AnomalyLikelihood likelihood_; //TODO which params/how pass them to constructor?
```

Comment:
> This approach, having members that are only used in some configurations, is not generally seen as good C++ design. 

So let's make it right! 
How about: 
- `Anomaly` interface with `virtual Real compute(...)`
- `AnomalyRaw : Anomaly` which is basically this class in the current state (w/o the likelihood_ field)
- `AnomalyLikelihood: AnomalyRaw` which will override the compute() ? 
- we could then get rid of the `AnomalyMode` is it would be inferred by the class inheritance
- I just don't know how to implement the mode `WRIGHTED` which is `raw()*likelihood()`, but maybe that can be left to the client side, and we'll only provide raw & likelihood here. 
@mrcslws shall I proceed with that?